packages/nimble-components/src/table/specs/table-column-custom-sort-field-hld.md

```diff
@@ -0,0 +1,57 @@
+# Custom table column sort order HLD
+
+## Problem Statement
+
+In some cases, a client may have the need to specify a custom sort order for a column. For example, a table displaying SystemLink tags may have a field in their records for the stringified tag but may not want to sort that column by that stringified value. Instead, they want to provide their own sort order for tags that takes each tag's data type and value into account.
+
+## Links to Relevant Work Items and Reference Material
+
+[Azdo Feature 2665131 - Custom column sorting](https://dev.azure.com/ni/DevCentral/_workitems/edit/2665131)
+
+## Implementation / Design
+
+Columns that need support for a custom column sort order will extend their API to allow a client to optionally specify an additional field within the data that specifies the sort order of that column. This field is expected to be a numeric value that is the sorted data index of the record. When the table is sorted by this column, the table will use the `basic` sort operation and sort the records by the additionally specified field rather than the default `operandDataRecordFieldName` associated with the column.
```

Comment:
> This field is expected to be a numeric value that is the sorted data index of the record

It doesn't actually have to be though right? It just has to be a set or orderable numbers (could be negative, could be fractional)? Maybe we can phrase it as a recommended pattern, but it's not required (and we don't depend on the data to actually be indexes right?)